scrapy/core/downloader/__init__.py

```diff
@@ -143,7 +143,7 @@ def _get_slot(self, request: Request, spider: Spider) -> Tuple[str, Slot]:
 
         return key, self.slots[key]
 
-    def _get_slot_key(self, request: Request, spider: Spider) -> str:
+    def _get_slot_key(self, request: Request, spider: Any) -> str:
```

Comment:
Not `Optional[Spider]`? It this because of tests passing junk, or something else?

Reply:
It doesn't really matter as the argument is not used. We should deprecate and remove it I think, we can't remove it right away because it's used both in private and public code (e.g. in `ScrapyZyteAPIDownloaderMiddleware`)

Reply:
https://github.com/scrapy/scrapy/issues/6340